src/common/directives/creditCardNumber.directive.js

```diff
@@ -0,0 +1,83 @@
+import angular from 'angular'
+import appConfig from 'common/app.config'
+import cruPayments from 'cru-payments'
+
+const directiveName = 'creditCardNumber'
+
+const creditCardNumberDirective = /* @ngInject */ function () {
+  return {
+    restrict: 'A',
+    require: '?ngModel',
+    link: function (scope, element, attrs, ngModelCtrl) {
+      if (!ngModelCtrl) {
+        return
+      }
+
+      ngModelCtrl.$parsers.unshift(function (creditCardNumber) {
+        let trimmedCardNum = creditCardNumber.replace(/\s+/g, '').replace(/\D/gi, '')
+        const cardType = cruPayments.creditCard.card.info.type(creditCardNumber)
+        const numbers = []
+
+        if (cardType === 'Visa' || cardType === 'MasterCard' || cardType === 'Discover') {
+          if (trimmedCardNum.length > 16) {
+            trimmedCardNum = trimmedCardNum.substr(0, 16)
+          }
+          const partitions = [4, 4, 4, 4]
+          let position = 0
```

Comment:
It could be nice to split this logic out into a separate function since it is used three times. We could even add tests for that method since it's not trivial logic.